### ISSUE TITLE
Include model instance arguements with weights

### ISSUE THREAD

**Issue by wfondrie** [Noble-Lab/casanovo#156]
Currently, users must know the architecture of a model in order to load the weights. It would be nice if these were included in the model weights that we release, so that the correct model could be automatically initialized.

If we fully rely on PyTorch lightning for checkpointing, it includes this information by default. From the [PyTorch Lightning docs](https://lightning.ai/docs/pytorch/latest/common/checkpointing_basic.html#contents-of-a-checkpoint):
> ## Contents of a checkpoint
> A Lightning checkpoint contains a dump of the model’s entire internal state. Unlike plain PyTorch, Lightning saves everything you need to restore a model even in the most complex distributed training environments.
> Inside a Lightning checkpoint you’ll find:
> - 16-bit scaling factor (if using 16-bit precision training)
> - Current epoch
> - Global step
> - LightningModule’s state_dict
> - State of all optimizers
> - State of all learning rate schedulers
> - State of all callbacks (for stateful callbacks)
> - State of datamodule (for stateful datamodules)
> - **The hyperparameters (init arguments) with which the model was created**
> - The hyperparameters (init arguments) with which the datamodule was created
> - State of Loops

This would make for a easier user experience.
